### PR TITLE
ceph-volume: add ceph.osdspec_affinity tag

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jschmid@suse.de>

Part 1) of https://tracker.ceph.com/issues/44755
Resolves: https://tracker.ceph.com/issues/44929   (only the DG_AFFINITY part)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/ceph-volume/ceph_volume/api/lvm.py [hunk 1]
```diff
@@ -946,6 +946,7 @@ def report(self):
                 'type': type_,
                 'osd_fsid': self.tags['ceph.osd_fsid'],
                 'cluster_fsid': self.tags['ceph.cluster_fsid'],
+                'drivegroup_affinity': self.tags.get('ceph.drivegroup_affinity', ''),
```
Comment (sebastian-philipp) [inline src/ceph-volume/ceph_volume/api/lvm.py:949]: hm. `s/drivegroup/orchestrator_osd_spec`?

Comment (jschmid1) [reply src/ceph-volume/ceph_volume/api/lvm.py:949]: probably, yes

**Comment by jschmid1** [comment]
We try to source this information in the orchestrators to map OSDs to their respective OSDSpec. It fells like an awful long way to query ceph-volume on each host to retrieve this information. It would probably be better off in the OSD's metadata (`ceph osd metadata`).

What do you think @jan--f @jdurgin ?

**Comment by jdurgin** [comment]
It seems reasonable to report this from the osd metadata - it could go in its own metadata file like fsid/device_class/etc.

File: src/ceph-volume/ceph_volume/api/lvm.py (hunk 1 above)
Comment (jschmid1) [reply src/ceph-volume/ceph_volume/api/lvm.py:949]: done

**Comment by jschmid1** [comment]
> It seems reasonable to report this from the osd metadata - it could go in its own metadata file like fsid/device_class/etc.

see https://github.com/ceph/ceph/pull/34835

**Comment by jschmid1** [comment]
test results: http://pulpito.ceph.com/jschmid-2020-04-30_14:45:11-rados-wip-jschmid1-testing-2020-04-30-1530-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
we need this backported to octopus only. I hope this will work out.

**Comment by jschmid1** [comment]
As long as we backport the related PRs at the same time and since we ship ceph-osd, which includes ceph-volume in one container, we should end up with compatible versions..

If we would need a version guard, we should implement this in cephadm I think..

**Comment by jan--f** [comment]
> As long as we backport the related PRs at the same time and since we ship ceph-osd, which includes ceph-volume in one container, we should end up with compatible versions..
> 
> If we would need a version guard, we should implement this in cephadm I think..

This is not just container related...most clusters out there are not containerized. And I doubt backporting the osd changes to mimic is feasible, is it?

**Comment by jschmid1** [comment]
> > As long as we backport the related PRs at the same time and since we ship ceph-osd, which includes ceph-volume in one container, we should end up with compatible versions..
> > If we would need a version guard, we should implement this in cephadm I think..
> 
> This is not just container related...most clusters out there are not containerized. And I doubt backporting the osd changes to mimic is feasible, is it?

Right, but you explicitly have to set the environment variable, which only cephadm implements currently.

**Comment by jan--f** [comment]
I opened https://tracker.ceph.com/issues/45374 for the blacklist part

**Pull request merged**